packages/cli/src/cog/cog.stac.job.ts

```diff
@@ -0,0 +1,416 @@
+        concurrency?: number;
+
+        /**
+         * Override the source projection
+         */
+        projection?: Epsg;
+
+        /**
+         * Resampling method
+         * @Default  GdalCogBuilderDefaults.resampling
+         */
+        resampling?: GdalCogBuilderResampling;
+    };
+
+    /**
+     * Should this job be submitted to batch now?
+     * @default false
+     */
+    batch: boolean;
+
+    /** Should this job ignore source coverage and just produce one big COG for EPSG extent */
+    oneCogCovering: boolean;
+}
+
+/**
+ * Attempt to parse a year from a imagery name
+ * @example wellington_urban_2017_0.10m -> 2017
+ * @param name Imagery name to parse
+ * @return imagery year, -1 for failure to parse
+ */
+export function extractYearRangeFromName(name: string): [number, number] {
+    const re = /(?:^|\D)(\d{4})(?:-(\d{2}))?(?:$|\D)/g;
+
+    const years: number[] = [];
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        years.push(parseInt(m[1]));
+        if (m[2] != null) years.push(parseInt(m[1].slice(0, 2) + m[2]));
+    }
+
+    if (years.length == 0) {
+        return [-1, -1];
+    }
+
+    years.sort();
+
+    return [years[0], years[years.length - 1] + 1];
+}
+
+/**
+ * Information needed to create cogs
+ */
+export class CogStacJob implements CogJob {
+    json: CogJobJson;
+    private cacheTargetZoom?: {
+        gsd: number;
+        zoom: number;
+    };
+
+    /**
+     * Load the job.json
+
+     * @param jobpath where to load the job.json from
+     */
+    static async load(jobpath: string): Promise<CogStacJob> {
+        return new CogStacJob(await FileOperator.readJson<CogJobJson>(jobpath));
+    }
+
+    /**
+     * Create job.json, collection.json, source.geojson, covering.geojson, cutlint.geojson.gz and
+     * stac descriptions of the target COGs
+
+     * @param id unqiue id for this imagery set
+     * @param imageryName of the imagery set
+     * @param metadata information about the source
+     * @param ctx information about the output
+     * @param cutlinePoly the polygon to use to clip the source imagery
+     * @param addAlpha do we need an alpha band
+     */
+    static async create({
+        id,
+        imageryName,
+        metadata,
+        ctx,
+        cutlinePoly,
+        addAlpha,
+    }: {
+        id: string;
+        imageryName: string;
+        metadata: CogBuilderMetadata;
+        ctx: JobCreationContext;
+        cutlinePoly: MultiPolygon;
+        addAlpha: boolean;
+    }): Promise<CogStacJob> {
+        let description: string | undefined;
+        const providers: StacProvider[] = [];
+        const links: StacLink[] = [
+            {
+                href: FileOperator.join(ctx.outputLocation.path, 'collection.json'),
+                type: 'application/json',
+                rel: 'self',
+            },
+            {
+                href: 'job.json',
+                type: 'application/json',
+                rel: 'job',
```

Comment:
could you be more specific, eg `linz:basemaps:job` or something 

Is there any sort of structure for `rel` and what is in/out?

Reply:
I'll rename `linz.basemaps.job`